File: sorting/selection.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

void selection(int arr[],int len){

    for (int i=0;i<len-1;i++){
        int imin=i;
        for(int j=i+1 ; j<len;j++){
            if(arr[j]<arr[imin])
            imin=j;
        }
        int temp=arr[i];
        arr[i]=arr[imin];
        arr[imin]=temp;
    }
    
}
int main(){

    int arr[]={2,1,6,5,7,3};
    int len=sizeof(arr)/sizeof(arr[1]);
    selection(arr,len);
    for (int i=0;i<len;i++)
        cout<<arr[i]<<" ";
}

File: sorting/selection.py
def selection(arr,l):
    for i in range(0,l-1):
        imin=i
        for j in range(i+1,l):
            if arr[j]<arr[imin]:
                imin=j
        arr[i],arr[imin]=arr[imin],arr[i]
    


if __name__ == "__main__":
    arr=[2,1,6,5,7,3]
    l=len(arr)
    selection(arr,l)
    print(arr)